File: apps/web/src/lib/materials/money.ts
// Para — .NET Money ile aynı kurallar. Varsayılan TRY; USD/EUR desteklenir. Float ile değil
// string/decimal-temsille taşınır (sınırda doğrulama). İşlem anı kur snapshot ayrı saklanır.
export const BASE_CURRENCY = "TRY";
const ALLOWED = new Set(["TRY", "USD", "EUR"]);

export const isSupportedCurrency = (c: string | null | undefined): boolean =>
  !!c && ALLOWED.has(c);

// Invariant ondalık metin (nokta ayraç). Geçersizse 0.
export function parseAmount(text: string | null | undefined): number {
  if (text === null || text === undefined || text.trim() === "") return 0;
  const v = Number(text);
  return Number.isFinite(v) ? v : 0;
}

export const serializeAmount = (value: number): string => String(value);

File: src/DepoWise.Application/Common/Money.cs
using System.Globalization;

namespace DepoWise.Application.Common;

/// <summary>
/// Para = decimal tutar + currency_code. Float KULLANILMAZ. Varsayılan TRY; USD/EUR desteklenir.
/// İşlem anındaki kur snapshot olarak ayrı saklanır (bu tip kuru taşımaz).
/// </summary>
public readonly record struct Money(decimal Amount, string Currency)
{
    public const string BaseCurrency = "TRY";
    private static readonly HashSet<string> Allowed = new(StringComparer.Ordinal) { "TRY", "USD", "EUR" };

    public static Money Try(decimal amount) => new(amount, BaseCurrency);

    public static bool IsSupported(string? currency) => currency is not null && Allowed.Contains(currency);

    public static string Serialize(decimal value) => value.ToString(CultureInfo.InvariantCulture);

    public static decimal Parse(string? text)
        => decimal.TryParse(text, NumberStyles.Number, CultureInfo.InvariantCulture, out var v) ? v : 0m;

    public string SerializeAmount() => Serialize(Amount);
}
